### PR TITLE
Add B3 headers injection and extraction

### PR DESCRIPTION
**Pull request by bwoebi**
### Description

Support for B3 headers.

### Readiness checklist
- [x] (only for Members) Changelog has been added to the release document.
- [x] Tests added for this feature/bug.

### Reviewer checklist
- [ ] Appropriate labels assigned.
- [ ] Milestone is set.
- [ ] Changelog has been added to the release document. For community contributors the reviewer is in charge of this task.

### CODE REVIEW

**Review by labbati — CHANGES_REQUESTED** [review]
Thanks @bwoebi for the work on B3 headers, I added a few comments or request for clarification.

File: ext/php7/handlers_curl.c [hunk 1]
```diff
@@ -141,15 +155,43 @@ static int dd_inject_distributed_tracing_headers(zval *ch) {
...
+            } else if (sampling_priority == PRIORITY_SAMPLING_USER_KEEP) {
+                b3_sampling_decision = "d";
+            } else {
+                b3_sampling_decision = "1";
+            }
+        }
+        if (trace_id && span_id) {
+            add_next_index_str(&headers, zend_strpprintf(0, "b3: %" PRIx64 "-%" PRIx64 "%s%s", trace_id, span_id, b3_sampling_decision ? "-" : "", b3_sampling_decision ? b3_sampling_decision : ""));
+        } else if (b3_sampling_decision) {
+            add_next_index_str(&headers, zend_strpprintf(0, "b3: %s", b3_sampling_decision));
```
[the remark this answers is not in the document]
Comment (bwoebi) [reply ext/php7/handlers_curl.c:191]:
```
A deny decision encodes as:

b3: 0
```
from https://github.com/openzipkin/b3-propagation#single-header

**Comment by bwoebi** [comment]
Thanks for the review, I'll apply the suggestions soon.

File: ext/php7/ddtrace.c [hunk 2]
```diff
@@ -1968,30 +1968,86 @@ zend_module_entry ddtrace_module_entry = {STANDARD_MODULE_HEADER_EX, NULL,
...
         DDTRACE_G(distributed_parent_trace_id) = ddtrace_parse_userland_span_id(&parent_zv);
+    } else if (DDTRACE_G(trace_id) && parse_b3 && zai_read_header_literal("X_B3_SPANID", &parent_id_str) == ZAI_HEADER_SUCCESS) {
+        zval parent_zv;
+        ZVAL_STR(&parent_zv, parent_id_str);
+        DDTRACE_G(distributed_parent_trace_id) = ddtrace_parse_hex_span_id(&parent_zv);
     }
 
-    DDTRACE_G(dd_origin) = NULL;
-    if (zai_read_header_literal("X_DATADOG_ORIGIN", &DDTRACE_G(dd_origin)) == ZAI_HEADER_SUCCESS) {
+    if (parse_datadog && zai_read_header_literal("X_DATADOG_ORIGIN", &DDTRACE_G(dd_origin)) == ZAI_HEADER_SUCCESS) {
```
[the remark this answers is not in the document]
Comment (labbati) [reply ext/php7/ddtrace.c:2068]: @bwoebi please remove `parse_datadog` or convince me of the opposite 😄

**Review by labbati — APPROVED** [review]
Thanks for the work on this!

**Pull request merged**